qiskit_ionq/helpers.py

```diff
@@ -296,27 +296,30 @@ def get_register_sizes_and_labels(registers):
     return sizes, labels
 
 
-def compress_dict_to_metadata_string(metadata_dict):  # pylint: disable=invalid-name
+def compress_to_metadata_string(metadata):  # pylint: disable=invalid-name
     """
-    Convert a dict to a compact string format (dumped, gzipped, base64 encoded)
+    Convert a list of dicts to a compact string format (dumped, gzipped, base64 encoded)
     for storing in IonQ API metadata
 
     Parameters:
-        metadata_dict (dict): a dict with metadata relevant to building the results
-            object on a returned job.
+        metadata (dict or list): a dict or list of dicts with metadata relevant
+            to building the results object on a returned job.
 
     Returns:
         str: encoded string
 
     """
-    serialized = json.dumps(metadata_dict)
+    serialized = json.dumps(metadata)
     compressed = gzip.compress(serialized.encode("utf-8"))
     encoded = base64.b64encode(compressed)
     encoded_string = encoded.decode()
+    encoded_string_length = len(encoded_string)
+    if encoded_string_length >= 40000:  # 40000 char is an IonQ API limitation
```

Comment:
slight preference for us doing this enforcement server-side only (notifying the user if the job submission is rejected) but I'm okay with this for now :)

Reply:
youre right we already raise an exception! ill remove this